monitoring/timedrift.go

```diff
@@ -273,16 +290,18 @@ func (c *timeDriftChecker) shouldCheckNode(node serf.Member) bool {
 func (c *timeDriftChecker) getAgentClient(ctx context.Context, node serf.Member) (client.Client, error) {
 	c.mu.Lock()
```

Comment:
Sorry this is not a part of your changeset but I'm just noticing that we're acquiring a lock here over DialRPC function which actually attempts to dial a remote server. This can potentially lead to the entire checker stall if DialRPC hangs. Since you're here anyway, could you please rewrite this so we only place locks on map operations and not I/O? I.e. the connection should be created outside the lock.